tests/conftest.py

```diff
@@ -401,7 +401,9 @@ def filter_heartbeat_events():
 
 @pytest.fixture
 def telemetry_writer():
```

Comment:
You could do something like:

```python
@pytest.fixture
def telemetry_writer_is_agentless():
    return False

@pytest.fixture
def telemetry_writer(telemetry_writer_is_agentless):
    telemetry_writer = TelemetryWriter(is_periodic=False, agentless=telemetry_writer_is_agentless)
```


This way in the future if we ever want an individual test to be agentless it'll be easy to overwrite the fixture value.

Reply:
not a strong opinion, an option/pattern we have used elsewhere.